app/models/action/edition_diff_factory.rb

```diff
@@ -0,0 +1,45 @@
+class Action::EditionDiffFactory
+  attr_reader :previous_draft_edition, :updated_draft_edition
+
+  def add_previous_draft_edition(previous_draft_edition)
+    @previous_draft_edition = get_edition_attributes(previous_draft_edition)
+  end
+
+  def add_updated_draft_edition(updated_draft_edition)
+    @updated_draft_edition = get_edition_attributes(updated_draft_edition)
+  end
+
+  def create_draft_edition_diff
+    # we don't raise an error if there is no @previous_draft_edition present,
+    # as it's possible for no previous draft edition to exist
+    raise Action::NoUpdatedDraftEditionProvidedError unless @updated_draft_edition.present?
```

Comment:
Might be that we require both methods to have been called but nil results in an empty object, so if either is nil it means it wasn't called and then can throw exception - could probably avoid the need for a comment then.